### ISSUE TITLE
Expose out-of-fold per-sample prediction scores (cross-validated predict + std) in AAPred

### ISSUE THREAD

**Issue by breimanntools** [breimanntools/aaanalysis#398]
## Problem

`AAPred.predict` scores samples with models fit on **all** the training data
(`aaanalysis/prediction/_backend/aa_pred/aa_pred_fit.py`: `fit_models` →
`.fit(X, labels)`, then `predict_proba_models`). Scoring the *training*
proteins this way is in-sample and optimistically biased. To get honest
per-protein scores for the training set, users fall back to hand-rolled
sklearn:

```python
P = np.vstack([cross_val_predict(m, X, y, cv=cv5, method="predict_proba")[:, 1] for m in ensemble])
score, std = P.mean(0), P.std(0)          # mean over the ensemble, std across models
```

In the γ-secretase notebook this exact block appears **twice** — once for the
substratome ranking (Figs 12–14) and again to build the SHAP prediction context
— duplicating the out-of-fold scoring that the prediction layer should own.

Note the gap is only half-open: `AAPred.predict` **already** returns `score`
and `score_std` (std across the ensemble) for *new* proteins. What is missing is
the **out-of-fold** (cross-validated) score for the *training* set.

## Goal

Expose cross-validated **out-of-fold per-sample scores** (mean over the
ensemble + std across models) for the training data through the public `AAPred`
API, reproducing the hand-rolled ensemble exactly and leaving the existing
new-protein `predict` output unchanged.

## Requirements

- [ ] A public entry point for out-of-fold scores — e.g. `AAPred.predict(..., cv=...)`,
      or a dedicated `AAPred.predict_oof(X, ...)` / a fitted `oof_scores_`
      attribute. Decide one shape and document it.
- [ ] Returns per-sample `score` + `score_std`, computed as **per-model**
      `cross_val_predict(method="predict_proba")` then aggregated across models
      (match the notebook's `np.vstack([...]).mean(0)` / `.std(0)` ordering, not
      aggregate-then-CV).
- [ ] Reuse the existing `predict_proba_models` aggregation shape (`score`,
      `score_std`) so the output columns match `predict`.
- [ ] Deterministic under `random_state` (StratifiedKFold shuffle-seed contract).
- [ ] Frontend validation block; backend trusts frontend; numpydoc with named
      `Returns` and an `Examples` include.

## KPIs / Acceptance criteria

- [ ] OOF `score`/`score_std` equal within `1e-9` to
      `np.vstack([cross_val_predict(m, X, y, cv=StratifiedKFold(5, shuffle=True, random_state=42), method="predict_proba")[:, 1] for m in ensemble])`
      aggregated (`mean(0)` / `std(0)`) across the 4-model ensemble.
- [ ] The γ-secretase Appendix-3 ensemble block **and** the SHAP proba block are
      each replaced by a single call; Figs 12–14 and the downstream SHAP /
      clustermap figures render **byte-identical**.
- [ ] Existing new-protein `predict` output **byte-identical** to current
      (regression-tested).
- [ ] Covered by ≥1 unit test and ≥1 reproducibility (seed) test.

## Scope / non-goals

- Core feature (`topic:core`); no new dependencies, no `pro` gating.
- Not the full paper-fidelity nested-CV Monte-Carlo training engine — that is
  #276; this is the minimal out-of-fold scoring primitive underneath it.
- No bootstrap confidence intervals — that is #91.

## Dependencies

- relates #276 (paper-fidelity training engine), #91 (repeated-CV / CIs).
- surfaced while reducing the γ-secretase Appendix-3 boilerplate (cf. #305).

## Standards checklist

- [ ] **CONFIRM-FIRST**: touches the public `AAPred` signature (additive,
      backward compatible) — flag for review.
- [ ] Frontend/backend split honored; validation block; backend trusts frontend.
- [ ] numpydoc docstring (named `Returns`, per-method `Examples` include).
- [ ] tests (unit + reproducibility); `random_state` contract.
- [ ] no `print()` (`ut.print_out`); bare `ValueError`/`RuntimeError`; no
      `aaanalysis._utils.*` imports outside `utils.py`.

**Comment by breimanntools** [comment]
Sibling issue: #397 (LeaveOneOut / custom-CV in `AAPred.eval`). Both were surfaced together while shrinking the γ-secretase use-case Appendix 3; each is independently done-able and output-preserving.
